File: 문제풀이/프로그래머스/코딩 기초 트레이닝/Level 0. 주사위 게임 2.py
def solution(a: int, b: int, c: int) -> int:
    answer = a + b + c

    if len({a, b, c}) <= 2:
        answer *= a**2 + b**2 + c**2
    if len({a, b, c}) <= 1:
        answer *= a**3 + b**3 + c**3

    return answer


print(solution(2, 6, 1))
print(solution(5, 3, 3))
print(solution(4, 4, 4))

File: 문제풀이/프로그래머스/코딩 기초 트레이닝/Level 0. 주사위 게임 2.js
function solution(a, b, c) {
  if (a === b && b === c && c === a) {
    return (a + b + c) * (a ** 2 + b ** 2 + c ** 2) * (Math.pow(a, 3) + Math.pow(b, 3) + Math.pow(c, 3));
  } else if (a !== b && b !== c && c !== a) {
    return a + b + c;
  } else if ((a === b && a !== c) || (b === c && b !== a) || (c === a && c !== b)) {
    return (a + b + c) * (a ** 2 + b ** 2 + c ** 2);
  }
}
